### PR TITLE
Add #6 Add hall of fame

### PR DESCRIPTION
**Pull request by ThulinaWickramasinghe**
- Add routing for hall-of-fame
- WIP dynamic data fetching for hof

### CODE REVIEW

File: src/store/api/leaderboard/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  reducerPath: "leaderboardApi",
+  baseQuery,
+  endpoints: (builder) => ({
+    fetchLeaderboard: builder.query({
+      query: ({ round, ghostLegion }) => `/api/leaderboard?round=${round}&ghost_legion=${ghostLegion}`,
+      transformResponse: transformLeaderboardData
+    }),
+    fetchPastLeaderboards: builder.query({
+      queryFn: async ({ year = 2023, round, ghostLegion, page, filters, sorts }) => {
+        let path = round == 1 ? "round1" : ghostLegion == true ? "ghost-legion" : "final";
```
Comment (Akalanka47000) [inline src/store/api/leaderboard/index.js:15]: Let's make this triple equals else the linter will fail

File: src/store/api/leaderboard/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        try {
+          const data = await import(`../../../constants/annual-leaderboard/${year}/${path}.json`);
+          const transformedData = transformLeaderboardData({ ...data.default }, undefined, { page, filters, sorts });
+          return { data: transformedData };
+        } catch (error) {
+          console.log("Failed to fetch data dynamically");
```
Comment (Akalanka47000) [inline src/store/api/leaderboard/index.js:22]: Lets change this to `console.error`

File: src/store/api/leaderboard/transformer.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { orderBy } from "lodash";
```
Comment (Akalanka47000) [inline src/store/api/leaderboard/transformer.js:1]: import this as `import { default as orderBy } from "lodash/orderBy";`

**Review by Akalanka47000 — CHANGES_REQUESTED** [review]

File: src/store/api/leaderboard/index.js (hunk 1 above)
Comment (ThulinaWickramasinghe) [reply src/store/api/leaderboard/index.js]: Done

File: src/store/api/leaderboard/transformer.js (hunk 3 above)
Comment (ThulinaWickramasinghe) [reply src/store/api/leaderboard/transformer.js]: Done

**Review by Akalanka47000 — CHANGES_REQUESTED** [review]

File: src/components/layout/header.jsx [hunk 4]
```diff
@@ -15,8 +17,14 @@ const buttonStyles = "mt-1.5 xl:mt-0 px-8 xl:px-[1.15rem] pb-2.5 xl:pb-[0.4rem]
 
 const Header = ({ className }) => {
   const [mobileNavOpen, setMobileNavOpen] = useState(false);
+  const [isHallOfFame, setIsHallOfFame] = useState();
 
   const breakpoints = useBreakpoint();
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    setIsHallOfFame(pathname === hallOfFame);
```
[the remark this answers is not in the document]
Comment (ThulinaWickramasinghe) [reply src/components/layout/header.jsx:26]: This is being used in 2 places. Will add to a variable and use it.

[the remark this answers is not in the document]
Comment (ThulinaWickramasinghe) [reply src/components/layout/header.jsx]: Done

**Review by Akalanka47000 — APPROVED** [review]

**Pull request merged**